app/models/game.rb

```diff
@@ -13,6 +13,25 @@ class Game < ActiveRecord::Base
 
   scope :available, -> { where('black_player_id IS NULL OR white_player_id IS NULL') }
 
+  def check?
+    #game is in check if either black or white king is in check
+    king_is_in_check?('black') || king_is_in_check('white')
+  end
+
+  def king_is_in_check?(color)
+    #finds king by color
+    king = pieces.find_by(type: "King", color: color)
+    #returns array of all opposing pieces
+    opponents = pieces.includes(:game).where(color: !king.color).to_a
+    #loops through each opposing piece to determine if they have a valid move
+    #that can capture king
+    opponents.each do |opponent|
+      if opponent.valid_move?(king.x, king.y) == true
+        return true
+      end
```

Comment:
this could be changed to 1 line. :smile: 

`return true if opponent.valid_move?(king.x, king.y)`